ESSArch_Core/storage/backends/tests/test_disk.py

```diff
@@ -0,0 +1,277 @@
+        disk_storage_backend = DiskStorageBackend()
+        mock_st_medium.storage_target.target = self.datadir
+        mock_st_method.containers = True
+        src_list = ["some_src", "some_src2", "some_src3"]
+
+        disk_storage_backend.write(
+            src=src_list,
+            ip=mock.ANY,
+            storage_method=mock_st_method,
+            storage_medium=mock_st_medium
+        )
+
+        expected_dest = self.datadir
+        expected_copy_calls = [mock.call(src, expected_dest, block_size=65536) for src in src_list]
+
+        self.assertEqual(mock_copy.call_count, 3)
+        mock_copy.assert_has_calls(expected_copy_calls)
+        mock_st_obj.assert_called_once()
+
+    @mock.patch("ESSArch_Core.ip.models.InformationPackage")
+    @mock.patch("ESSArch_Core.storage.backends.disk.copy")
+    @mock.patch("ESSArch_Core.storage.models.StorageObject.objects.create")
+    @mock.patch("ESSArch_Core.storage.models.StorageMethod")
+    @mock.patch("ESSArch_Core.storage.models.StorageMedium")
+    def test_write_src_with_multi_files_not_container(self, st_medium, mock_st_method, mock_st_obj, mock_copy, ip):
+        disk_storage_backend = DiskStorageBackend()
+        st_medium.storage_target.target = self.datadir
+        mock_st_method.containers = False
+        ip.object_identifier_value = "some/object/path"
+        src_list = ["some_src", "some_src2", "some_src3"]
+
+        disk_storage_backend.write(
+            src=src_list,
+            ip=ip,
+            storage_method=mock_st_method,
+            storage_medium=st_medium
+        )
+
+        expected_dest = os.path.join(self.datadir, "some/object/path")
+        expected_copy_calls = [mock.call(src, expected_dest, block_size=65536) for src in src_list]
+
+        self.assertEqual(mock_copy.call_count, 3)
+        mock_copy.assert_has_calls(expected_copy_calls)
+        mock_st_obj.assert_called_once()
+
+    @mock.patch("ESSArch_Core.storage.models.StorageObject")
+    @mock.patch("ESSArch_Core.storage.backends.disk.copy")
+    def test_read_not_container_should_call_copy(self, mock_copy, mock_storage_obj):
+        disk_storage_backend = DiskStorageBackend()
+
+        mock_storage_obj.container = False
+        mock_storage_obj.get_full_path.return_value = "the_full_path"
+
+        disk_storage_backend.read(
+            storage_object=mock_storage_obj,
+            dst="some_dest"
+        )
+
+        mock_copy.assert_called_once_with("the_full_path", "some_dest", block_size=65536)
+
+    @mock.patch("ESSArch_Core.storage.models.StorageObject")
+    @mock.patch("ESSArch_Core.ip.models.InformationPackage")
+    @mock.patch("ESSArch_Core.storage.backends.disk.copy")
+    def test_read_container_default(self, mock_copy, mock_ip, mock_storage_obj):
+        disk_storage_backend = DiskStorageBackend()
+        full_path_to_src = "the/full/path/to/src"
+        container_path = os.path.join("container/path")
```

Comment:
`os.path.join` has no effect here

```suggestion
        container_path = "container/path"
```